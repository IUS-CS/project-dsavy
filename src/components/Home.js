import React from 'react';
import { Container, Row, Col } from 'react-bootstrap'
import ClickableImage from './Home-image';
import Card from './Card.js'

function Home() {
  return (
      <Container className="container-fluid d-flex justify-content-center">
        <Row>
            <Col>
                <Card name="Linked Lists" image="../images/LinkedList.png" page='/linkedlist'
                    description="A linear data structure in which a node has a value and a pointer to next node"/>
            </Col>
            <Col>
                <Card name="Stacks" image="../images/LinkedList.png" />
            </Col>
            <Col>
                <Card name="Queues" image="../images/LinkedList.png" />
            </Col>
        </Row>

        {/* <Row style={{padding:10}}>
          <Col><ClickableImage name='Linked List' image='../images/LinkedList.png' page='/linkedlist'/></Col>
          <Col><ClickableImage name='Stack' image='../images/LinkedList.png' page='/about'/></Col>
          <Col><ClickableImage name='Data Structure' image='../images/LinkedList.png' page='/about'/></Col>
        </Row>
        <Row>
        <Col><ClickableImage name='Linked List' image='../images/LinkedList.png' page='/about'/></Col>
          <Col><ClickableImage name='Linked List' image='../images/LinkedList.png' page='/about'/></Col>
          <Col><ClickableImage name='Linked List' image='../images/LinkedList.png' page='/about'/></Col>
        </Row> */}
      </Container>
  );
}

export default Home;