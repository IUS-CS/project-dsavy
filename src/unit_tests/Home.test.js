import React from 'react';
import Home from 'components/Home';

test("returns correct component", () => {
    expect(Home()).toStrictEqual(<div>This is the Home Component</div>)
});


