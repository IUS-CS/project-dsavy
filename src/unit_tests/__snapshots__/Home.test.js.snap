// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<div
  className="container-sm"
>
  <div
    className="row"
    style={
      Object {
        "paddingTop": 40,
      }
    }
  >
    <div
      className="col"
    >
      <a
        href="/linkedlist"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="custom-card card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top"
            src="../images/LinkedList.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Linked List
            </div>
            <p
              className="card-text"
            >
              Text...
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      className="col"
    >
      <a
        href="/stack"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="custom-card card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top"
            src="../images/Stack.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Stack
            </div>
            <p
              className="card-text"
            >
              Text...
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      className="col"
    >
      <a
        href="/queue"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="custom-card card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top"
            src="../images/LinkedList.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Queue
            </div>
            <p
              className="card-text"
            >
              Text...
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "paddingTop": 40,
      }
    }
  >
    <div
      className="col"
    >
      <a
        href="/about"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="custom-card card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top"
            src="../images/LinkedList.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Linked List
            </div>
            <p
              className="card-text"
            >
              Text...
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      className="col"
    >
      <a
        href="/about"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="custom-card card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top"
            src="../images/LinkedList.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Linked List
            </div>
            <p
              className="card-text"
            >
              Text...
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      className="col"
    >
      <a
        href="/about"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="custom-card card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            className="card-img-top"
            src="../images/LinkedList.png"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Linked List
            </div>
            <p
              className="card-text"
            >
              Text...
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
